### ISSUE TITLE
Weird tab behaviour on newline

### ISSUE THREAD

**Issue by dario-loi** [sitandr/conchord#2]
# The Problem
Currently, the way that tabulatures split up on newlines is not pretty at all, since most of the times measures get broken up into multiple lines, significantly degrading readability.

## Examples

I tried my hand at a [transcription](https://gist.github.com/dario-loi/73973c89f7ebdb07de09c2b5621033da) of a song from italian band Gomma, (original available on Ultimate Guitar [here](https://tabs.ultimate-guitar.com/tab/gomma/come-va-paolo-tabs-4684385), and the song is really not legible if I don't manually escape the newlines every four measures:

![paolo_white](https://github.com/sitandr/conchord/assets/77234595/7b9be46d-222c-41d6-9a21-e31c5e62df15)
*Here we don't have weird breaks, however, sixteenth notes get smushed together when they are fretted on a double digit fret, so they are not very legible*

A solution to the smushing of the notes would be to increase the scale of the tabulatures, but then:

![paolo_white_2](https://github.com/sitandr/conchord/assets/77234595/d411515e-9b31-408b-8b51-7e4ede57012a) 
*The measures get broken up across lines and it becomes difficult to tell the duration of the notes (I also removed the manual linebreaks to accentuate this issue)*

## Solutions (?)

Newlines could trigger the addition of a blank measure that fills up the remaining space from the last one to the end of the current line, so that the next true measure begins and ends on the next line. Also, to fix the smushing (which frankly should be its separate issue), there should be some padding applied to the number in the measure so that they don't touch eachother.

**Comment by sitandr** [comment]
Thanks a lot for your feedback!

Unfortunately, weird tab behaviour on breaks is the thing I wanted to fix, but the task is quite hard, so I didn't want to spend my time on this if no one would need it anyway. Now when I see someone needs it, I'm going to so more work on this.

So yes, unfortunately, right now everywhere manual line breaks are required.

Now I think I will look into this, but, unfortunately, I have quite a sparse free time right now. If it's not urgent for you, I can't guarantee I will make it this year. But I definitely want to work on it right after NY.

Speaking about smashed notes, do you think it could be a good idea to reduce text `size` if there is not enough space? That should be quite easy to implement.

**Comment by dario-loi** [comment]
Yup, reduction of text size in tabs and support for the "b" character would be a great patch of both issue #1 and the current one.

I was just checking out the package and providing feedback, I don't need the features/fixes to be pushed urgently, so by all means take your time and thanks for your work!

**Comment by sitandr** [comment]
Sorry for the delay. I made some large overhaul of how the tab part of the library works. Now it makes several layout attempts and tries to adjust the bar size (this is similar to most common behaviour on real tablature editors). It also now auto-breaks on bar lines.

I also implemented scaling to make notes not to smash together:

![image](https://github.com/sitandr/conchord/assets/60141933/d0594b0a-7def-4b49-a9c6-b0be926e5e6d)

I think I can close this issue now. I hope to publish an update for the package soon!
